dataflow/src/ops/trigger.rs

```diff
@@ -0,0 +1,223 @@
+use std::collections::HashMap;
+
+use prelude::*;
+
+use hyper::Client;
+use tokio_core::reactor::Core;
+use serde::Serialize;
+use serde_json;
+use std::thread;
+
+/// Applies the identity operation to the view. Since the identity does nothing,
+/// it is the simplest possible operation. Primary intended as a reference
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Trigger {
+    us: Option<IndexPair>,
+    src: IndexPair,
+    trigger: TriggerType,
+    key: Vec<usize>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum TriggerType {
+    // url to the blender, group name
+    GroupCreation{ url: String, group: String },
+}
+
+impl Trigger {
+    /// Construct a new Trigger operator.
+    pub fn new(src: NodeIndex, trigger: TriggerType, key: Vec<usize>) -> Trigger {
+        assert_eq!(key.len(), 1);
```

Comment:
If it has to be length 1, why not take `key: usize`?

Reply:
I imagine that if we want to trigger other events that are not just `GroupCreation` we might want a larger set of keys. Although for now they are all of length 1, since that's the only event we support.

Reply:
Then I think I'd just make it a `usize` for now, and we can worry about extending it further later. The extra heap allocation and `[0]`s aren't worth it.